test/e2e/initialize.sh

```diff
@@ -42,3 +42,37 @@ if [[ "$NETWORK_STATUS" != "healthy" ]]; then
 fi
 
 $soroban keys generate $SOROBAN_ACCOUNT
+
+# retrieve the contracts using soroban contract init then build them if they dont already exist
+# Define directory and WASM file paths
+target_dir="$dirname/test-contracts/target/wasm32-unknown-unknown/release"
+wasm_files=(
+    "soroban_custom_types_contract.wasm"
+    "soroban_atomic_swap_contract.wasm"
+    "soroban_token_contract.wasm"
+    "soroban_increment_contract.wasm"
+    "hello_world.wasm"
+)
+
+# Check if all WASM files exist
+all_exist=true
+for wasm_file in "${wasm_files[@]}"; do
+    if [ ! -f "$target_dir/$wasm_file" ]; then
+        all_exist=false
+        break
+    fi
+done
+
+# If any WASM file is missing, initialize and build the contracts
```

Comment:
Should we bother with this check? Even if they exist they might be out of date from the repo, or is that not possible / not worth worrying about?

Reply:
You are correct, they could be out of date. My thought was that if they are you could just delete them manually and have them update since its probably pretty rare for that to be an issue. What I could do is check the git hash and update them if they mismatch.